src/app/components/loaders/stx20-tokens-loader.tsx

```diff
@@ -24,13 +24,23 @@ function getTokenId(token: Stx20TokenItem) {
   return token.info.symbol;
 }
 
+function castFullTokenInfo(rawToken: Partial<Stx20CryptoAssetInfo>) {
+  return {
+    chain: rawToken.chain!,
+    category: rawToken.category!,
+    protocol: rawToken.protocol!,
+    symbol: rawToken.symbol!,
+    decimals: rawToken.decimals!,
+    hasMemo: rawToken.hasMemo!,
+  };
+}
```

Comment:
What are your thoughts on using a zod validator here to ensure the type?

Reply:
A schema validator could for example, filter the array for those that do not match, proving runtime safety

Reply:
We do control data access here via our own library call `useStx20Tokens`, so we shouldn't really need any runtime type validation.

The reason this was typed as a Partial in `@leather.io/query` was to accommodate the recent [model refactor](https://github.com/leather-io/mono/pull/565#discussion_r1829762108) that (temporarily) adds two unused fields to `Stx20CryptoAssetInfo`.

To support that change we either needed to assert once like this or change every downstream file/component that handles `Stx20CryptoAssetInfo` to handle partials, which would have been a big increase in scope.